### PR TITLE
Enable android module if present when creating Flutter project

### PR DESCRIPTION
**Pull request by skybrian**
If "flutter create" writes an android.iml file, make sure it's added to the project.

Also, make sure all files generated by the "flutter create" command are visible in IDEA. (There was a race condition where we didn't wait for "flutter create" to finish before continuing.)

Progress towards fixing #782
The android.iml file will be added in https://github.com/flutter/flutter/pull/9335

### CODE REVIEW

**Comment by alexander-doroshko** [comment]
Make sure to test in small IDEs (no android support + unexpected things may break in multi-modular projects there). Also makes sense to test in IntelliJ IDEA with disabled Android plugin.

File: src/io/flutter/module/FlutterModuleBuilder.java [hunk 1]
```diff
@@ -89,22 +156,40 @@ public ModuleType getModuleType() {
...
+  private static void runFlutterCreate(@NotNull FlutterSdk sdk, Module module, @NotNull VirtualFile baseDir)
+    throws ConfigurationException {
     // Create files.
     try {
-      sdk.run(FlutterSdk.Command.CREATE, model.getModule(), baseDir, null, baseDir.getPath());
+      final Process process =
+        sdk.startProcess(FlutterSdk.Command.CREATE, module, baseDir, null, baseDir.getPath());
+      if (process != null) {
+        // Wait for process to finish. (We overwrite some files, so make sure we lose the race.)
+        // TODO(skybrian) the user sees a short pause. Show progress somehow?
```
Comment (devoncarew) [inline src/io/flutter/module/FlutterModuleBuilder.java]: Is this running in the wizard dialog, or in message console in the new IntelliJ window?

**Comment by devoncarew** [comment]
> Make sure to test in small IDEs (no android support + unexpected things may break in multi-modular projects there). Also makes sense to test in IntelliJ IDEA with disabled Android plugin.

Re: that comment, I think where we want to get to with this PR is for the Android code in a Flutter project to be editable in IDEA, and for Dart code to continue to work in small IDEs (WebStorm) w/o issues (i.e., it's not an active use case for the Android code to work in WebStorm, but we don't want exceptions during project creation there).

**Comment by pq** [comment]
> > Make sure to test in small IDEs (no android support + unexpected things may break in multi-modular projects there). Also makes sense to test in IntelliJ IDEA with disabled Android plugin.

>Re: that comment, I think where we want to get to with this PR is for the Android code in a Flutter project to be editable in IDEA, and for Dart code to continue to work in small IDEs (WebStorm) w/o issues (i.e., it's not an active use case for the Android code to work in WebStorm, but we don't want exceptions during project creation there).

A quick (if not long-term) solution might be to simply skip the android module configuration in all but the IDEA case.  That should solve the IDEA problem and ensure we don't confuse the small-IDEs with unsupported multi-module projects.

Thoughts?

File: src/io/flutter/module/FlutterModuleBuilder.java (hunk 1 above)
Comment (skybrian) [inline src/io/flutter/module/FlutterModuleBuilder.java]: For IDEA this runs after the project wizard disappears and before the new project window appears - not ideal.

For WebStorm, the new project window is open, but blank.

**Comment by skybrian** [comment]
WebStorm project creation takes a different code path, so the android module isn't used. (I renamed a method to make this clearer.)

When I create a project in WebStorm, the Dart SDK is not set correctly. However, it seems that was already the case without this patch.

**Pull request merged**